server/src/main/java/io/crate/expression/scalar/string/ReverseFunction.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Licensed to Crate.io GmbH ("Crate") under one or more contributor
+ * license agreements.  See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.  Crate licenses
+ * this file to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial agreement.
+ */
+
+package io.crate.expression.scalar.string;
+
+import io.crate.data.Input;
+import io.crate.metadata.Functions;
+import io.crate.metadata.NodeContext;
+import io.crate.metadata.Scalar;
+import io.crate.metadata.TransactionContext;
+import io.crate.metadata.functions.BoundSignature;
+import io.crate.metadata.functions.Signature;
+import io.crate.types.DataTypes;
+
+public final class ReverseFunction extends Scalar<String, Object> {
+
+    private static final String NAME = "reverse";
+
+    public static void register(Functions.Builder module) {
+        module.add(
+            Signature.scalar(
+                NAME,
+                DataTypes.STRING.getTypeSignature(),
+                DataTypes.STRING.getTypeSignature()
+
+            ),
+            ReverseFunction::new
+        );
+    }
+
+    public ReverseFunction(Signature signature, BoundSignature boundSignature) {
+        super(signature, boundSignature);
+    }
+
+    @Override
+    public String evaluate(TransactionContext txnCtx, NodeContext nodeCtx, Input<Object>[] args) {
+        assert args.length == 1 : "reverse takes exactly one arguments";
+        var text = (String) args[0].value();
+
+        if (text == null) {
+            return null;
+        }
+
+        return new StringBuffer(text).reverse().toString();
+    }
```

Comment:
The StringBuilder is more efficient than the StringBuffer.
Also we can just use the `UnaryScalar` for this.

```suggestion
            (signature, boundSignature) -> {
                return new UnaryScalar<>(
                    signature,
                    boundSignature,
                    DataTypes.STRING,
                    s -> new StringBuilder(s).reverse().toString()
                )
            }
        );
    }
```